src/contrasts.jl

```diff
@@ -224,6 +231,8 @@ for contrastType in [:DummyCoding, :EffectsCoding, :HelmertCoding, :SeqDiffCodin
     end
 end
 
+baselevel(c::C) where {C<:AbstractContrasts} = hasfield(C, :base) ? c.base : nothing
```

Comment:
Yeah it is hacky.  But you can always provide a more specific method than this for other types so the user-facing API wouldn't change at all.

Reply:
Yeah, but then if one day you wanted to rename this field, it could break external implementations. That's not very likely, but better avoid exposing internal fields.

Reply:
Ah I see now.  Yes better to have an accessor for the types that have the `base` field and a separate fallback...